DOCSUP-6073: Documented mapContains, mapKeys, mapValues functions (#20325)
ClickHouse/ClickHouse: docs/en/sql-reference/functions/tuple-map-functions.md

└──────────────────────────────┴───────────────────────────────────┘
```

## mapContains {#mapcontains}

> Determines  whether `map.keys` contains the `key` parameter.

Reviewer: ```suggestion
Determines  whether the `Map` contains the `key` parameter.
```

**Syntax**

``` sql
mapContains(map, key)
```

**Parameters** 

-   `map` — Map. 

> -   `key` — Key. Type matches the type of `map.keys`.

Reviewer: ```suggestion
-   `key` — Key. Type matches the type of `key` parameter.
```

**Returned value**

> -   `1` if `map.keys` contains `key`, `0` if not.

Reviewer: ```suggestion
-   `1` if `Map` contains `key`, `0` if not.
```

Type: [UInt8](../../sql-reference/data-types/int-uint.md).

**Example**

Query:

```sql
CREATE TABLE test (a Map(String,String)) ENGINE = Memory;

INSERT INTO test VALUES ({'name':'eleven','age':'11'}), ({'number':'twelve','position':'6.0'});

SELECT mapContains(a, 'name') FROM test;

```

Result:

```text
┌─mapContains(a, 'name')─┐
│                      1 │
│                      0 │
└────────────────────────┘
```

## mapKeys {#mapkeys}

> Returns all the keys from `map` parameter.

Reviewer: ```suggestion
Returns all keys from the `map` parameter.
```

**Syntax**

```sql
mapKeys(map)
```

**Parameters**

-   `map` — Map.

**Returned value**

> -   Array containing all the keys from `map`.

Reviewer: ```suggestion
-   Array containing all keys from the `map`.
```

Type: [Array](../../sql-reference/data-types/array.md).

**Example**

Query:

```sql
CREATE TABLE test (a Map(String,String)) ENGINE = Memory;

INSERT INTO test VALUES ({'name':'eleven','age':'11'}), ({'number':'twelve','position':'6.0'});

SELECT mapKeys(a) FROM test;
```

Result:

```text
┌─mapKeys(a)────────────┐
│ ['name','age']        │
│ ['number','position'] │
└───────────────────────┘
```

## mapValues {#mapvalues}

> Returns all the values from `map` parameter.

Reviewer: ```suggestion
Returns all values from the `map` parameter.
```
